WordPress/src/main/java/org/wordpress/android/ui/mediapicker/MediaItem.kt

```diff
@@ -13,24 +14,21 @@ data class MediaItem(
     val mimeType: String? = null,
     val dataModified: Long
 ) {
-    sealed class Identifier {
-        data class LocalUri(val value: UriWrapper) : Identifier()
-        data class RemoteId(val value: Long) : Identifier()
-        data class StockMediaIdentifier(val url: String?, val name: String?, val title: String?) : Identifier()
+    sealed class Identifier : Parcelable {
+        @Parcelize
+        data class LocalUri(val value: @RawValue UriWrapper) : Identifier()
 
-        fun toParcel() = Parcel((this as? LocalUri)?.value?.uri, (this as? RemoteId)?.value)
+        @Parcelize
+        data class RemoteId(val value: Long) : Identifier()
 
-        companion object {
-            fun fromParcel(parcel: Parcel): Identifier {
-                return when {
-                    parcel.remoteId != null -> RemoteId(parcel.remoteId)
-                    parcel.uri != null -> LocalUri(UriWrapper(parcel.uri))
-                    else -> throw IllegalArgumentException("Parcel doesn't have URI or remote ID")
-                }
-            }
-        }
+        @Parcelize
+        data class StockMediaIdentifier(val url: String?, val name: String?, val title: String?) : Identifier()
 
         @Parcelize
-        data class Parcel(val uri: Uri? = null, val remoteId: Long? = null) : Parcelable
+        data class GifMediaIdentifier(
+            val mediaModel: MediaModel?,
```

Comment:
I was thinking about this. Do you think it might make sense to split the `GifMediaIdentifier` into 2 classes, one without the `MediaModel` and the other called something like `LocalId` (we already have `RemoteId`)? And map one to the other in the Insert use case? It might even work to change the functionality from using `LocalId` to `RemoteId` since we're using that for the other cases (even stock library). This would probably require a little more work (and we can do that in a separate PR) but might be a good step towards consistency.